packages/subgraph/src/mappings/cfav1.ts

```diff
@@ -37,13 +47,104 @@ function createFlowUpdatedEntity(
     ev.stream = streamId;
     ev.totalAmountStreamedUntilTimestamp = totalAmountStreamedUntilTimestamp;
 
-    let type = oldFlowRate.equals(BIG_INT_ZERO)
+    let type = getFlowActionType(oldFlowRate, event);
+    ev.type = type;
+    ev.save();
+}
+
+function getFlowActionType(
+    oldFlowRate: BigInt,
+    event: FlowUpdated
+): FlowActionType {
+    return oldFlowRate.equals(BIG_INT_ZERO)
         ? FlowActionType.create
         : event.params.flowRate.equals(BIG_INT_ZERO)
         ? FlowActionType.terminate
         : FlowActionType.update;
-    ev.type = type;
-    ev.save();
+}
+
+function handleStreamPeriodUpdate(
+    event: FlowUpdated,
+    previousFlowRate: BigInt,
+    stream: Stream
+): void {
+    let streamRevision = getOrInitStreamRevision(
+        event.params.sender.toHex(),
+        event.params.receiver.toHex(),
+        event.params.token.toHex()
+    );
+    let flowActionType = getFlowActionType(previousFlowRate, event);
+    let previousStreamPeriod = StreamPeriod.load(
+        getStreamPeriodID(stream.id, streamRevision.periodRevisionIndex)
+    );
+    switch (flowActionType) {
+        case FlowActionType.create:
+            startStreamPeriod(event, streamRevision, stream.id);
+            break;
+        case FlowActionType.update:
+            if (!previousStreamPeriod) {
+                throw "Previous StreamPeriod not found for flow terminate action";
```

Comment:
"update" instead of "terminate".

Also let's use:
`throw Error("Previous StreamPeriod not found for flow terminate action");` as it's the so-called better practice: https://stackoverflow.com/questions/11502052/throwing-strings-instead-of-errors